docs/azure/sdk/authentication/local-development-broker.md

```diff
@@ -0,0 +1,123 @@
+
+- **Azure Blob Storage**: Assign a role such as **Storage Account Data Contributor**.
+- **Azure Key Vault**: Assign a role such as **Key Vault Secrets Officer**.
+
+If an app is specified, it must have API permissions set for **user_impersonation Access Azure Storage** (step 6 in the previous section). This API permission allows the app to access Azure storage on behalf of the signed-in user after consent is granted during sign-in.
+
+## Implement the code
+
+Complete the following steps to use <xref:Azure.Identity.InteractiveBrowserCredential> in a MAUI app to authenticate with the [`SecretClient`](/dotnet/api/azure.security.keyvault.secrets.secretclient):
+
+1. Install the [Azure.Identity](https://www.nuget.org/packages/Azure.Identity) and [Azure.Identity.Broker](https://www.nuget.org/packages/Azure.Identity.Broker) packages.
+
+    > [!NOTE]
+    > macOS and Linux support exists in `Azure.Identity.Broker` versions 1.3.0 and later.
+
+1. Get a reference to the parent window on top of which the account picker dialog should appear.
+1. Create an instance of <xref:Azure.Identity.InteractiveBrowserCredential> that accepts an instance of <xref:Azure.Identity.Broker.InteractiveBrowserCredentialBrokerOptions>.
+
+:::zone target="docs" pivot="os-windows"
+
+:::code language="csharp" source="../snippets/authentication/brokered/maui-app/MainPage.xaml.cs" highlight="37-49" :::
+
+The following screenshot shows the user sign-in experience:
+
+:::image type="content" source="../media/web-account-manager-sign-in-account-picker.png" alt-text="A screenshot that shows the sign-in experience when using a broker-enabled InteractiveBrowserCredential instance to authenticate a user." :::
+
+:::zone-end
+
+:::zone target="docs" pivot="os-macos"
+
+:::code language="csharp" source="../snippets/authentication/brokered/maui-app/MainPage.xaml.cs" highlight="53-65" :::
+
+:::zone-end
+
+:::zone target="docs" pivot="os-linux"
+
+:::code language="csharp" source="../snippets/authentication/brokered/console-app/Program.cs" highlight="22-28" :::
+
+:::zone-end
+
+### Authenticate the default system account
+
+`InteractiveBrowserCredential` also supports a silent sign-in process that automatically uses a default account so the user doesn't have to repeatedly select it. Once you opt in to this behavior, the credential attempts to sign in by asking the underlying Microsoft Authentication Library (MSAL) to perform the sign-in for the default system account. If the sign-in fails, the credential falls back to displaying the account picker dialog, from which the user can select the appropriate account.
+
+The previous example shows how to enable sign-in with the default system account:
+
+:::code language="csharp" source="../snippets/authentication/brokered/console-app/Program.cs" range="22-28" :::
```

Comment:
I think this part can be reused for all 3 languages to simplify the doc.

Reply:
Are you suggesting using a console app for all 3 OSes?